src/core/graph.ts

```diff
@@ -424,6 +447,43 @@ export class DependencyGraph<T> {
 export class SortableDependencyGraph<
   T extends Sortable<number>
 > extends DependencyGraph<T> {
+  static fromDependencyGraph<R, T extends Sortable<number>>(
```

Comment:
This practice is very Java-y, but I don't think using a constructor would be a good idea as ts doesn't allow overloading so that will cause some headache.